samcli/commands/check/lambda_function_pricing_calculations.py

```diff
@@ -0,0 +1,227 @@
+"""
+Class for calculating pricing information for all lambda functiuons
+"""
+
+from typing import List
+import urllib.request
+from urllib.error import HTTPError
+
+import ast
+
+from botocore.session import get_session
+
+from samcli.commands.check.pricing_calculations import PricingCalculations
+
+
+class LambdaFunctionPricingCalculations(PricingCalculations):
+    def __init__(self, graph):
+        super().__init__(graph)
+        self._max_num_of_free_requests = None
+        self._max_free_gb_s = None
+        self._monthly_compute_charge = None
+        self._monthly_request_charge = None
+        self._max_request_usage_type = "Global-Request"
+        self._max_free_gb_s_usage_type = "Global-Lambda-GB-Second"
+        self._compute_usage_type = "Lambda-GB-Second"
+        self._request_charge_usage_type = "Request"
+
+        self.lambda_pricing_results = None
+
+    def get_aws_pricing_info(self, region):
+        return _get_aws_lambda_pricing_info(region)
+
+    def determine_cost(self) -> None:
+        self._determine_lambda_cost()
```

Comment:
This is the same as one of the above comments - it looks redundant to me and I think you can implement what `_determine_lambda_cost` contains directly under `determine_cost`. 

Aside - I understand that `determine_cost` is part of the interfaces inherited from `PricingCalculations`, which is good 👍.